src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        return calculate(names, data, dateFrom, dateTo);
+    }
+
+    public LocalDate parseData(String input) {
+        final String pattern_data = "dd.MM.yyyy";
+        try {
+            DateTimeFormatter formatter =
+                    DateTimeFormatter.ofPattern(pattern_data);
+            LocalDate date = LocalDate.parse(input, formatter);
+            return date;
+        } catch (DateTimeParseException exc) {
+            System.out.printf("%s is not parsable!%n", input);
+            throw exc;
+        }
+    }
+
+    public String calculate(String[] names, String[] data, String dateFrom, String dateTo) {
+        StringBuilder calculation = new StringBuilder();
+        calculation.append("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo);
+        for (String inputNames : names) {
```

Comment:
```suggestion
        for (String name: names) {
```
we are working with a single value here